### PR TITLE
docs: catch actual exception in java doc comment

### LINKED ISSUE #309: Don't recommend catching java.lang.Exception
https://googleapis.dev/java/google-cloud-storage/latest/com/google/cloud/storage/Blob.html#writer-com.google.cloud.storage.Storage.BlobWriteOption...-

sample code includes a catch block for java.lang.Exception. Per Effective Java et al, this is an antipattern. The sample code should catch only the actual exceptions expected to be thrown.

### PR DESCRIPTION
**Pull request by suraj-qlogic**
Fixes #309

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 1]
```diff
@@ -728,7 +728,7 @@ public ReadChannel reader(BlobSourceOption... options) {
    * try (WriteChannel writer = blob.writer()) {
```
Comment (elharo) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:728]: no need for the nested try block, one will do here

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java [hunk 2]
```diff
@@ -2505,7 +2505,7 @@ Blob create(
    * try (WriteChannel writer = storage.writer(blobInfo)) {
```
Comment (elharo) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java:2505]: don't nest the try block, one is enough

Comment (suraj-qlogic) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java:2505]: Done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java (hunk 1 above)
Comment (suraj-qlogic) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:728]: Done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 3]
```diff
@@ -726,12 +726,10 @@ public ReadChannel reader(BlobSourceOption... options) {
    * <pre>{@code
    * byte[] content = "Hello, World!".getBytes(UTF_8);
    * try (WriteChannel writer = blob.writer()) {
-   *   try {
    *     writer.write(ByteBuffer.wrap(content, 0, content.length));
-   *   } catch (Exception ex) {
+   *   } catch (IOException ex) {
```
Comment (frankyn) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:730]: The end `}` isn't aligned with the opening try(). Please fix the spacing. This should be done below as well.

Comment (suraj-qlogic) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: Done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 4]
```diff
@@ -726,11 +726,9 @@ public ReadChannel reader(BlobSourceOption... options) {
...
    * try (WriteChannel writer = blob.writer()) {
-   *   try {
    *     writer.write(ByteBuffer.wrap(content, 0, content.length));
-   *   } catch (Exception ex) {
-   *     // handle exception
-   *   }
+   * } catch (IOException ex) {
+   *   // handle exception
    * }
    * }</pre>
```
Comment (elharo) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:733]: you still have an extra closing brace

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java [hunk 5]
```diff
@@ -2503,11 +2503,9 @@ Blob create(
...
-   *   try {
    *     writer.write(ByteBuffer.wrap(content, 0, content.length));
-   *   } catch (Exception ex) {
-   *     // handle exception
-   *   }
+   * } catch (IOException ex) {
+   *   // handle exception
    * }
    * }</pre>
```
Comment (elharo) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java:2510]: extra close brace

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java (hunk 4 above)
Comment (suraj-qlogic) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:733]: @elharo ,I have gone throught code and found there is no extra closing brace.Here last closing 
brace indicate closing of {@code } tag.Can you please look once?

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java (hunk 5 above)
Comment (suraj-qlogic) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java:2510]: Same as above.

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java (hunk 4 above)
Comment (elharo) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:733]: line 733, just before </pre>, same below

Comment (suraj-qlogic) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:733]: @elharo 
```
   * <pre>{@code
   * byte[] content = "Hello, World!".getBytes(UTF_8);
   * try (WriteChannel writer = blob.writer()) {
   *     writer.write(ByteBuffer.wrap(content, 0, content.length));
   * } catch (IOException ex) {
   *   // handle exception
   * }
   * }</pre>
```
Here First closing ```}``` brace indicate closing of try block,Second } brace indicate closing 
for catch clause.and last one ```}``` indicate closing of ```<pre>{@code``` so can we write document comment without ending ```<pre>{@code```?

Comment (elharo) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:733]: By my count you have two open braces and three close braces. Change line 733 from

`}</pre>` to `</pre>`

Comment (suraj-qlogic) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: Done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 6]
```diff
@@ -726,13 +726,11 @@ public ReadChannel reader(BlobSourceOption... options) {
...
-   *   try {
    *     writer.write(ByteBuffer.wrap(content, 0, content.length));
-   *   } catch (Exception ex) {
-   *     // handle exception
-   *   }
+   * } catch (IOException ex) {
+   *   // handle exception
    * }
-   * }</pre>
+   * </pre>
```
Comment (frankyn) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java:733]: You do need this `}</pre>` closing bracket to close `{@code... }` on line 726

@elharo 
dpyxdV2b95J

Comment (suraj-qlogic) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: Done

**Pull request merged**